src/hooks/ticker.ts

```diff
@@ -0,0 +1,164 @@
+  bestBidQuantity: number
+  bestAskPrice: number
+  bestAskQuantity: number
+  openPrice: number
+  highPrice: number
+  lowPrice: number
+  totalTradedBaseAssetVolume: number
+  totalTradedQuoteAssetVolume: number
+  statisticsOpenTime: number
+  statisticsCloseTime: number
+  firstTradeId: number
+  lastTradeId: number
+  totalNumberOfTrades: number
+}
+
+export const useTokenPairTicker = (tokenPair: TokenPair, connectOnMount: boolean, retryCount = 3) => {
+  const retriesRemaining = useRef(retryCount)
+
+  // Chrome always returns an event code of 1006 so we have to keep track whether we closed the
+  // connection manually or not (1000)
+  const isManualClose = useRef(false)
+
+  // Use a ref instead of state so we mark the connection as closed immediately without
+  // triggering a re-render
+  const isConnected = useRef(false)
+  const [stream, setStream] = useState<TickerStream>(null)
+  const websocket = useRef<WebSocket>(null)
+
+  const connect = useCallback(() => {
+    if (navigator.onLine && !isConnected.current) {
+      const ws = new WebSocket(`wss://stream.binance.com:9443/ws/streams/${tokenPair}@ticker`)
+
+      ws.onopen = () => {
+        isConnected.current = true
+      }
+
+      ws.onclose = () => {
+        isConnected.current = false
+
+        if (retriesRemaining.current > 0 && !isManualClose.current && !isConnected.current) {
+          setTimeout(() => {
+            connect()
+            retriesRemaining.current -= 1
+          }, 5000)
+        }
+      }
+
+      ws.onerror = (error) => {
+        console.error('Ticker:', error)
+      }
+
+      ws.onmessage = (evt) => {
+        try {
+          const data = JSON.parse(evt.data) as StreamData
+
+          if (isConnected.current) {
+            setStream({
+              eventType: data.e,
+              eventTime: data.E,
+              symbol: data.s,
+              priceChange: parseFloat(data.p),
+              priceChangePercent: parseFloat(data.P),
+              weightAveragePrice: parseFloat(data.w),
+              firstTrade: parseFloat(data.x),
+              lastPrice: parseFloat(data.c),
+              lastQuantity: Number(data.Q),
+              bestBidPrice: parseFloat(data.b),
+              bestBidQuantity: Number(data.B),
+              bestAskPrice: parseFloat(data.a),
+              bestAskQuantity: Number(data.A),
+              openPrice: parseFloat(data.o),
+              highPrice: parseFloat(data.h),
+              lowPrice: parseFloat(data.l),
+              totalTradedBaseAssetVolume: Number(data.v),
+              totalTradedQuoteAssetVolume: Number(data.q),
+              statisticsOpenTime: Number(data.O),
+              statisticsCloseTime: Number(data.C),
+              firstTradeId: Number(data.F),
+              lastTradeId: Number(data.L),
+              totalNumberOfTrades: Number(data.n),
+            })
+          }
+        } catch (error) {
+          console.error(`Error parsing data from stream`, error)
```

Comment:
Not sure how consumer uses it, but we just `console.error()` the error here and nothing else it will keep old data in the `stream`. Might be not an issue if next event comes fine but if there are a lot of errors for some reason stream might get really out of date.
Not sure if its something that needs fixing, gotta check the consumer code too, but just thinking out loud.